kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala

```diff
@@ -579,6 +579,23 @@ object KyuubiConf {
       .toSequence
       .createWithDefault(Nil)
 
+  val ENGINE_DEREGISTER_JOB_MAX_FAILURES: ConfigEntry[Int] =
+    buildConf("engine.deregister.job.max.failures")
+      .doc("Number of failures of job before deregistering the engine.")
+      .version("1.2.0")
+      .intConf
+      .checkValue(_ > 0, "must be positive number")
+      .createWithDefault(4)
+
+  val ENGINE_DEREGISTER_EXCEPTION_TTL: ConfigEntry[Long] =
+    buildConf("engine.deregister.exception.ttl")
+      .doc("TTL of specified exceptions to deregister the engine. If there is no exception for a" +
+        " TTL, the exception should has been recovered and set its failure number to 0.")
```

Comment:
1？

Reply:
hmm, for the implementation, I just delay the judge for ttl until the next failure.
In fact, it is 0.

Reply:
`Time to live(TTL) for exceptions pattern specified in ${ENGINE_DEREGISTER_EXCEPTION_CLASSES.key} and ${ENGINE_DEREGISTER_EXCEPTION_MESSAGES.key} to deregister engines. Once the total error count hits the ${ENGINE_DEREGISTER_JOB_MAX_FAILURES.key} within the TTL, an engine will deregister itself and wait to self terminated. Otherwise, we suppose that the engine has recovered from temporary failures.`

Reply:
how about that? looks more user friendly to me, not the mechanism itself

Reply:
thanks, updated